javatests/arcs/core/storage/CapabilitiesResolverTest.kt

```diff
@@ -0,0 +1,140 @@
+        assertThat(resolver.createStorageKey(Capabilities.TiedToArc) is VolatileStorageKey).isTrue()
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities.TiedToRuntime)
+        }
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities.Persistent)
+        }
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities(setOf(Capabilities.Capability.TiedToArc, Capabilities.Capability.Persistent)))
+        }
+    }
+
+    @Test
+    fun capabilitiesResolver_createsStorageKeysCtor() {
+        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val resolver = CapabilitiesResolver(options,
+            mutableMapOf("ramdisk" to CapabilitiesResolver.CapabilitiesCreator(
+                Capabilities.TiedToRuntime, { (arcId), _ -> RamDiskStorageKey(arcId.toString()) }
+            ))
+        )
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities.TiedToArc)
+        }
+        assertThat(resolver.createStorageKey(Capabilities.TiedToRuntime) is RamDiskStorageKey).isTrue()
+    }
+
+    @Test
+    fun capabilitiesResolver_createsStorageKeys() {
+        RamDisk.clear()
+        DatabaseDriverProvider.configure(MockDatabaseFactory(), mapOf<String, Schema>()::get)
+        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val resolver1 = CapabilitiesResolver(options)
+        assertThat(resolver1.findStorageKeyProtocols(Capabilities.TiedToArc)).containsExactly("volatile")
+        assertThat(resolver1.findStorageKeyProtocols(Capabilities.TiedToRuntime)).containsExactly("ramdisk")
+        assertThat(resolver1.findStorageKeyProtocols(Capabilities.Persistent)).containsExactly("db")
+        assertThat(resolver1.createStorageKey(Capabilities.TiedToArc) is VolatileStorageKey).isTrue()
+        assertThat(resolver1.createStorageKey(Capabilities.TiedToRuntime) is RamDiskStorageKey).isTrue()
+        assertThat(resolver1.createStorageKey(Capabilities.Persistent, "abc012") is DatabaseStorageKey).isTrue()
+
+        CapabilitiesResolver.reset()
+        val resolver2 = CapabilitiesResolver(options)
+        assertThat(resolver2.createStorageKey(Capabilities.TiedToArc) is VolatileStorageKey).isTrue()
+        assertThrows(IllegalStateException::class) {
+            resolver2.createStorageKey(Capabilities.TiedToRuntime)
+        }
+    }
+
+    @Test
+    fun capabilitiesResolver_staticCreators() {
+        assertThat(CapabilitiesResolver.defaultCreators.size).isEqualTo(1);
+        assertThat(CapabilitiesResolver.registeredCreators.isEmpty()).isTrue()
+
+        CapabilitiesResolver.registerDefaultKeyCreator(
+            "test1",
+            Capabilities.TiedToRuntime,
+            {(arcId), _ -> RamDiskStorageKey(arcId.toString())}
```

Comment:
Move lambda to after the closing paren. Also - same destructuring nit as well.

Reply:
done